inference-engine/src/vpu/graph_transformer/src/frontend/detect_network_batch.cpp

```diff
@@ -82,94 +63,66 @@ void FrontEnd::detectNetworkBatch(
+                return;
+            }
 
-        switch (ieData->getLayout()) {
-        case ie::Layout::NCDHW:
-        case ie::Layout::NDHWC:
-        case ie::Layout::NCHW:
-        case ie::Layout::NHWC:
-        case ie::Layout::NC:
-        case ie::Layout::CN:
             ieShapes[0] = 1;
-            break;
-        default:
-            VPU_THROW_FORMAT("Input %v has unexpected layout %v", ieData->getName(), ieData->getLayout());
         }
-
         inputShapes[ieData->getName()] = ieShapes;
     }
 
+    if (inputBatch == 1) {
+        env.log->trace("Network is batch 1. Not need to reshape");
+        return;
+    }
+
+    if (inputBatch == 0) {
+        env.log->trace("Unable to decide on network batch size.");
+        return;
+    }
+    model->setBatchSize(checked_cast<int>(inputBatch));
+
     //
     // Special case for DetectionOutput.
     //
 
-    if (!network.getFunction()) {
-        for (auto it = ie::details::CNNNetworkIterator(network); it != ie::details::CNNNetworkIterator(); ++it) {
-            const auto& layer = *it;
-
-            if (!cmp(layer->type, "DetectionOutput")) {
-                continue;
-            }
-
-            env.log->trace("Found DetectionOutput layer [%s]", layer->name);
-
-            if (layer->outData.empty()) {
-                VPU_THROW_FORMAT("Unsupported layer %s configuration: no outputs", layer->name);
-            }
-
-            // 1. Don't support if DetectionOutput is not the last layer in network
-            if (!ie::getInputTo(layer->outData.front()).empty()) {
-                VPU_THROW_FORMAT("Unsupported layer %s configuration : it is not a network output", layer->name);
-            }
+    const auto layers = network.getFunction()->get_ordered_ops();
+    for (const auto& layer : layers) {
+        if (!ngraph::as_type_ptr<ngraph::op::DetectionOutput>(layer))
+            continue;
+        env.log->trace("Found DetectionOutput layer [%s]", layer->get_name());
 
-            // 2. Don't support if there multiple outputs as well
-            if (outputsInfo.size() != 1) {
-                VPU_THROW_FORMAT("Unsupported network configuration : layer %s must be the only output of the network", layer->name);
-            }
+        VPU_THROW_UNLESS(layer->get_output_size() > 0, "Unsupported layer {} configuration: no outputs", layer->get_name());
 
-            model->attrs().set<bool>("withDetectionOutput", true);
-        }
-    } else {
-        const auto layers = network.getFunction()->get_ops();
-        for (const auto& layer : layers) {
-            if (layer->get_type_name() != std::string("DetectionOutput"))
-                continue;
-
-            env.log->trace("Found DetectionOutput layer [%s]", layer->get_name());
-
-            if (layer->get_output_size() == 0)
-                VPU_THROW_FORMAT("Unsupported layer %s configuration: no outputs", layer->get_name());
-
-            // 1. Don't support if DetectionOutput is not the last layer in network
```

Comment:
Why did you remove this check?

Reply:
This check has been replaced by 
```
VPU_THROW_UNLESS(op->get_output_size() == 1, "Unsupported layer {} with type {} must have only 1 output but {} provided", op->get_name(), op->get_type_name(), op->get_output_size());
```

Reply:
It checks a number of layer outputs, not network outputs, doesn't it?